### PR TITLE
Add suffixes to Project Settings properties

### PR DESCRIPTION
**Pull request by samdze**
This PR aims to add all the right suffixes to Projects Settings properties.
Editor Settings suffixes can be addressed in a future PR.

Note that, working on this addition, I made a few changes to some property names, as now the unit of measurement is displayed by the suffix (and often also by the description):
- `application/run/frame_delay_msec` to `application/run/frame_delay`
- `audio/buses/channel_disable_threshold_db` to `audio/buses/channel_disable_threshold`
- `audio/video/video_delay_compensation_ms` to `audio/video/video_delay_compensation`
- `rendering/vulkan/staging_buffer/block_size_kb` to `rendering/vulkan/staging_buffer/block_size`
- `rendering/vulkan/staging_buffer/max_size_mb` to `rendering/vulkan/staging_buffer/max_size`
- `rendering/vulkan/staging_buffer/texture_upload_region_size_px` to `rendering/vulkan/staging_buffer/texture_upload_region_size`
- `gui/timers/incremental_search_max_interval_msec` to `gui/timers/incremental_search_max_interval`

I changed every occurrence of them in the source code, and also updated the `project_converter_3_to_4.cpp` file.

Also, there are a few settings I was not sure which suffix to add, these are:
- `physics/2d/sleep_threshold_linear`
- `physics/2d/sleep_threshold_angular`
- `physics/2d/default_gravity`
- `physics/2d/default_gravity_vector`
- `physics/2d/default_linear_damp`
- `physics/2d/default_angular_damp`
- `physics/2d/solver/contact_recycle_radius` (px?)
- `physics/2d/solver/contact_max_separation` (px?)
- `physics/2d/solver/contact_max_allowed_penetration` (px?)
- `physics/3d/sleep_threshold_linear`
- `physics/3d/sleep_threshold_angular`
- `physics/3d/default_gravity` (m/s?)
- `physics/3d/default_gravity_vector` (m/s?)
- `physics/3d/default_linear_damp`
- `physics/3d/default_angular_damp`
- `physics/3d/solver/contact_recycle_radius` (m?)
- `physics/3d/solver/contact_max_separation` (m?)
- `physics/3d/solver/contact_max_allowed_penetration` (m?)
- `network/limits/packet_peer_stream/max_buffer_po2`
- `rendering/environment/ssao/fadeout_from` (m?)
- `rendering/environment/ssao/fadeout_to` (m?)
- `rendering/environment/ssil/fadeout_from` (m?)
- `rendering/environment/ssil/fadeout_to` (m?)
- `rendering/environment/volumetric_fog/volume_size`
- `rendering/lightmapping/bake_performance/region_size` (px?)
- `rendering/mesh_lod/lod_change/threshold_pixels`
- `navigation/2d/default_cell_size`
- `navigation/2d/default_edge_connection_margin`
- `navigation/3d/default_cell_size`
- `navigation/3d/default_edge_connection_margin`

I also don't know if the naming change above is desired, so I'm leaving this PR as a draft to get feedback and fix stuff as things are decided.

### CODE REVIEW

File: doc/translations/ar.po [hunk 1]
```diff
@@ -31113,7 +31113,7 @@ msgid ""
 "select the item in question directly. Both of these actions will be reset to "
 "the beginning of the list if the timeout duration has passed since the last "
 "keystroke was registered. You can adjust the timeout duration by changing "
-"[member ProjectSettings.gui/timers/incremental_search_max_interval_msec]."
+"[member ProjectSettings.gui/timers/incremental_search_max_interval]."
```
Comment (aaronfranke) [inline doc/translations/ar.po:31116]: The `.po` and `.pot` files in the `doc/translations/` folder are auto-generated when syncing translations from Weblate, you should not manually change them.

**Comment by aaronfranke** [comment]
> Also, there are a few settings I was not sure which suffix to add, these are:

* `physics/2d/default_gravity`: `suffix:px/s\u00B2` (px/s²)
* `physics/3d/default_gravity`: `suffix:m/s\u00B2` (m/s²)

Take a look at the existing suffixes on the nodes. For example, look at Area and RigidDynamicBody for the physics ones. I looked at Area's gravity to confirm the above. For many of these you may need to dig into the code to find out the unit, but it's OK if you skip what you're unsure of for now. I also went hunting for the sleep threshold ones and found this line of code:

```cpp
	if (Math::abs(angular_velocity) < get_space()->get_body_angular_velocity_sleep_threshold() && Math::abs(linear_velocity.length_squared()) < get_space()->get_body_linear_velocity_sleep_threshold() * get_space()->get_body_linear_velocity_sleep_threshold()) {
```

Which told me that these are the same units as the angular velocity and linear velocity on RigidDynamicBody:

- `physics/2d/sleep_threshold_linear`: `suffix:px/s` (px/s)
- `physics/2d/sleep_threshold_angular`: `suffix:rad/s` (rad/s)
- `physics/3d/sleep_threshold_linear`: `suffix:m/s` (m/s)
- `physics/3d/sleep_threshold_angular`: `suffix:rad/s` (rad/s)

**Comment by aaronfranke** [comment]
> I also don't know if the naming change above is desired

I think we shouldn't change this in this PR. It may be desired to keep the old names, in case users view the project settings in a text editor instead of Godot's graphical settings menu. However, adding the suffixes, without changing the naming, is a very safe change that everyone can agree is good.

**Comment by reduz** [comment]
I think this is good and we can take the chance to do this including the renames. Let us know if there is interest in finishing it.

**Comment by samdze** [comment]
So, it seems renames are ok.
Yes, I'll get back to it soon.

**Comment by akien-mga** [comment]
One drawback of this approach is that while the unit will be clearly visible in the inspector, it won't be in the docs.

We could however parse the hint string and include units optionally in the class reference I think (in a separate PR).

**Comment by akien-mga** [comment]
Sadly this is now too late to do those compat breaking changes so close to the 4.0 release.

Adding the suffixes could still be added in a new PR, but the properties shouldn't be renamed.

**Pull request closed without merging**